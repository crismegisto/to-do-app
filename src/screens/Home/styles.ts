import { StyleSheet } from 'react-native';

const DEFAULT_SPACE = 20;

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  buttonContainer: {
    position: 'absolute',
    height: DEFAULT_SPACE * 2,
    width: '100%',
    bottom: DEFAULT_SPACE,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default styles;
