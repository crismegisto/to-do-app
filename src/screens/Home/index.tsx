import React, { useEffect, useState } from 'react';
import { View, FlatList, ListRenderItem, SafeAreaView } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import type { RootState } from 'src/store';
import { useSelector } from 'react-redux';
import CustomButton from 'components/CustomButton';
import TaskCard from 'components/TaskCard';
import { Task } from 'interfaces/task';
import { ListKeyExtractor } from 'interfaces/global';
import { filterByLabel } from 'utils/tasksFilters';

import { RoutesParamList } from 'constants/routesParamList';
import Routes from 'constants/routes';

import styles from './styles';

function Home() {
  const LABEL = 'work';
  const [filterTasks, setFilterTasks] = useState<Task[]>([]);
  const navigation = useNavigation<NativeStackNavigationProp<RoutesParamList>>();
  const tasks = useSelector((state: RootState) => state.tasks.currentList);

  useEffect(() => {
    const newTasks = filterByLabel(tasks, LABEL);
    setFilterTasks(newTasks);
  }, [tasks]);

  const renderItem: ListRenderItem<Task> = ({ item }) => <TaskCard {...item} />;

  const keyExtractor: ListKeyExtractor<Task> = ({ id }) => String(id);

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        renderItem={renderItem}
        keyExtractor={keyExtractor}
        data={filterTasks}
        extraData={filterTasks}
      />
      <View style={styles.buttonContainer}>
        <CustomButton name="Add Task" onSubmit={() => navigation.navigate(Routes.Detail)} />
      </View>
    </SafeAreaView>
  );
}

export default Home;
