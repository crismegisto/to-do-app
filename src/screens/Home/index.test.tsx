import React from 'react';
import { screen, fireEvent } from '@testing-library/react-native';

import { renderWithProvider, initialStoreState } from '../../../utils/test/redux';
import Home from './index';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate
    })
  };
});

describe('<CustomButton /> Tests', () => {
  it('match snapshot and button name', async () => {
    renderWithProvider(<Home />, {
      preloadedState: initialStoreState
    });

    expect(screen.toJSON()).toMatchSnapshot();

    const nameTask = screen.getByText('Mock');
    expect(nameTask).toBeOnTheScreen();

    const deleteButton = screen.getAllByTestId('deleteButton');
    fireEvent.press(deleteButton[0]);

    expect(nameTask).not.toBeOnTheScreen();
  });
});
