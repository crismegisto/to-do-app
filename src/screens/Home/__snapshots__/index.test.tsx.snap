// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> Tests match snapshot and button name 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "completed": false,
          "description": "",
          "id": 1,
          "labels": [
            "family",
            "work",
          ],
          "name": "Mock",
        },
        {
          "completed": false,
          "description": "",
          "id": 2,
          "labels": [
            "work",
          ],
          "name": "Example",
        },
      ]
    }
    extraData={
      [
        {
          "completed": false,
          "description": "",
          "id": 1,
          "labels": [
            "family",
            "work",
          ],
          "name": "Mock",
        },
        {
          "completed": false,
          "description": "",
          "id": 2,
          "labels": [
            "work",
          ],
          "name": "Example",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "borderWidth": 2,
              "flex": 1,
              "flexDirection": "row",
              "margin": 10,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              {
                "padding": 5,
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="checkBox"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    undefined,
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "marginHorizontal": 10,
                }
              }
              testID="name-list"
            >
              Mock
            </Text>
            <View
              data={
                [
                  "family",
                  "work",
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            >
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <Text>
                  family
                </Text>
              </View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <Text>
                  work
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="deleteButton"
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../src/assets/trash.png",
                }
              }
              style={
                {
                  "height": 24,
                  "tintColor": "red",
                  "width": 24,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "borderWidth": 2,
              "flex": 1,
              "flexDirection": "row",
              "margin": 10,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              {
                "padding": 5,
              }
            }
          >
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                {
                  "checked": false,
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "text": "off",
                }
              }
              accessible={true}
              pointerEvents="auto"
              testID="checkBox"
            >
              <RNCCheckbox
                forwardedRef={null}
                onValueChange={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "height": 32,
                      "width": 32,
                    },
                    undefined,
                  ]
                }
                value={false}
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "marginHorizontal": 10,
                }
              }
              testID="name-list"
            >
              Example
            </Text>
            <View
              data={
                [
                  "work",
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            >
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <Text>
                  work
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="deleteButton"
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../src/assets/trash.png",
                }
              }
              style={
                {
                  "height": 24,
                  "tintColor": "red",
                  "width": 24,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 20,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Add Task
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
